### PR TITLE
Nuke the Fluent theme

### PR DESCRIPTION
**Pull request by Jahnp**
#### Pull request checklist

- [ ] Addresses an existing issue: N/A (internal discussion)
- [x] Include a change request file using `$ yarn change`

#### Description of changes
The Fluent theme hasn't been necessary on `master` for some time now, as all styles are Fluent by default; its contents were really only meant for using the Fluent styles on v6. In addition, all of the Fluent work being done presently is either in `master` or in the Fluent theme on `fabric-6`.

This PR removes the Fluent theme from `master`, and is just cleanup for repo health—it doesn't affect any components or styles.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10372)

### CODE REVIEW

**Comment by Jahnp** [comment]
@phkuo FYI

File: change/@uifabric-experiments-2019-09-05-16-38-24-jahnp-nuke-fluent-theme.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "minor",
+  "comment": "Update references to Fluent theme in experiments  webpack configs",
```
Comment (cliffkoh) [inline change/@uifabric-experiments-2019-09-05-16-38-24-jahnp-nuke-fluent-theme.json]:
```suggestion
  "comment": "Remove redundant references to Fluent theme in experiments webpack config.",
```

File: change/@uifabric-experiments-2019-09-05-16-38-24-jahnp-nuke-fluent-theme.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "minor",
```
Comment (cliffkoh) [inline change/@uifabric-experiments-2019-09-05-16-38-24-jahnp-nuke-fluent-theme.json]: Might not be important but I think this actually has 0 functional effect on experiments, so can be `none`

Comment (Jahnp) [reply change/@uifabric-experiments-2019-09-05-16-38-24-jahnp-nuke-fluent-theme.json]: Ahh, very good callout—thanks @cliffkoh . Will adjust.

**Comment by Jahnp** [comment]
@cliffkoh thanks for reviewing! FYI for you and any other reviewers, the yarn file changed because the packages that reference the @uifabric/fluent-theme package now pull from npm registry instead of locally from the repo.

**Comment by ecraig12345** [comment]
> Not too sure why yarn lock file changed

yarn.lock changed because it's now pulling in the fluent-theme package from the registry due to some packages still depending on it (mainly for constants). @Jahnp Does the removal of this package mean that *no Fabric packages in master* should be using fluent-theme, or is it still okay for them to pull the last published version from the registry?

**Comment by ecraig12345** [comment]
Oops I belatedly read Peter's comment which points out the same thing as my comment, sorry! But it still seems a bit weird to have packages still pulling from one of our old packages which no longer exists in the repo.

**Comment by Jahnp** [comment]
@ecraig12345 sorry for the delay!

>Does the removal of this package mean that no Fabric packages in master should be using fluent-theme, or is it still okay for them to pull the last published version from the registry?

Good question. IMO, given that it's mostly the website that still uses the Fluent theme today (plus example-app-base for the page components), and given that it's reasonable to *document* the theme on the site, it's reasonable to keep the references but pull from the npm registry.

**Comment by ecraig12345** [comment]
@Jahnp I guess my concern is that if we delete this package (but don't officially deprecate it), then for some reason we need to update anything about it, that becomes very difficult. 

Also, I wonder if the constants the website uses ought to move to the styling package instead? (that's an actual question--I'm not sure if it's something we should do)

**Comment by Jahnp** [comment]
@ecraig12345 
> my concern is that if we delete this package (but don't officially deprecate it), then for some reason we need to update anything about it, that becomes very difficult.

That's a *very* fair point. While I don't foresee any reason to to update the 7.x version of the package, if something *did* come up, deleting the code would make that awkward. Not sure about the benefit of that vs a simpler/cleaner codebase, but I could definitely be persuaded to halt this for now.

In retrospect, seems like the right thing to do would have been to remove the package with the repo-wide 7.x release, before any @uifabric/fluent-theme@7.x releases happened. Maybe we save that for the 8.x release and in the meantime, simply deprecate the fluent-theme package on master. 

Thoughts?

**Comment by phkuo** [comment]
But we don't even maintain the 7.x version of fluent-theme. It is already, by definition, buggy. (For instance, it's missing command bar styles, which lead to very obvious visual regressions in sp-client.) The correct fix for consumers is to consume Fabric 6's version of the package, which is maintained and up to date.

If the downstream repo is on Fabric 7 and cannot consume Fabric 6 packages for some reason, then the correct thing to do is for them is to stop consuming Fabric 7's fluent-theme, because 7.x fluent-theme with Fabric 7 code is an unsupported and never intended scenario.

**Comment by ecraig12345** [comment]
I think I'd be okay with deleting the theme from the repo if we can find a new home for the constants (either `styling` if we're okay with other people consuming them, or `example-app-base` if we don't want to encourage consumption).

We'll also need someone with permissions to the npm account to manually mark `@uifabric/fluent-theme` as deprecated--@dzearing maybe?

**Comment by phkuo** [comment]
Yeah I don't want to encourage people to use them.
IMO we should start a new package that has nothing but constants and sample themes, which would give us an official way of shipping themes, like all the out of box themes SharePoint has developed. It'd be a good place to host product-specific themes too (like for Azure or Word).

**Comment by Jahnp** [comment]
Thanks for the discussion folks! I agree with the consensus to generally not encourage usage of the `styling` constants for now, but in the meantime, the shortest path to removing this theme is to replace them with either local values or constants from `styling`. I actually have that change locally with a failing build I'm investigating—would like to proceed with it for now if we're otherwise ok with removing the package in the repo + deprecating on npm. Would love @dzearing 's thoughts :)

@phkuo, that theme package sounds nifty—like someone a lot of folks would use. I could see there being a lot of value in a catalog of "official" themes like the ones you mention.

**Comment by ecraig12345** [comment]
Can we close this for now and reopen once we've decided on which path to take (and someone has time to do it)?

**Comment by Jahnp** [comment]
@ecraig12345 yup, that works for me—sorry to keep this open so long.

**Pull request closed without merging**